cylc/flow/parsec/fileparse.py

```diff
@@ -239,27 +259,54 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
         flines = inline(
             flines, fdir, fpath, False, viewcfg=viewcfg, for_edit=asedit)
 
+    # If empy and jinja2 sections are both filled raise an error.
+    if (all([extra_vars['empy:suite.rc'], extra_vars['jinja2:suite.rc']])):
+        raise FileParseError(
+            "Your additional configuration files define both empy and jinja2"
+            "variables. This makes sense."
+        )
+
+    template_vars['CYLC_VERSION'] = __version__
+
     # process with EmPy
     if do_empy:
+        if (
+            extra_vars['empy:suite.rc'] is not None and
+            not re.match(r'^#![Ee]m[Pp]y\s*', flines[0])
+        ):
+            flines.insert(0, '#!empy')
         if flines and re.match(r'^#![Ee]m[Pp]y\s*', flines[0]):
             LOG.debug('Processing with EmPy')
+            tvars = copy(template_vars)
+            if extra_vars['empy:suite.rc'] is not None:
+                for key, value in extra_vars['empy:suite.rc'].items():
+                    tvars[key] = value
             try:
                 from cylc.flow.parsec.empysupport import empyprocess
             except (ImportError, ModuleNotFoundError):
                 raise ParsecError('EmPy Python package must be installed '
                                   'to process file: ' + fpath)
-            flines = empyprocess(flines, fdir, template_vars)
+            flines = empyprocess(flines, fdir, tvars)
 
     # process with Jinja2
     if do_jinja2:
+        if (
+            extra_vars['jinja2:suite.rc'] is not None and
+            not re.match(r'^#![jJ]inja2\s*', flines[0])
+        ):
+            flines.insert(0, '#!jinja2')
```

Comment:
This is good, might be an idea to add an extra check here to make sure that some other shebang (e.g. empy) hasn't been defined: along the lines of:

```python
if re.match('^#!'. flines[0]):
    raise Exception()
```